src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/utils/validation.ts

```diff
@@ -384,6 +384,20 @@ function attachmentsValid(attachments: any, component: any): boolean {
   );
 }
 
+export function getRuleBasedDate(date: string, defaultDate?: string): moment.Moment | string {
```

Comment:
If you remove `moment.Moment | ` from here and return and add `as string` on line 399
You don't need the `as string` on line 88 & 89 in DatepickerComponent.tsx

Reply:
Actually it seems that it makes more sense to return moment.Moment type here.
I do however think the condition for returning a defaultDate is kind of weird.
As the defaultDate is optional.
I think you might as well not have it as an argument and return `null` regardless.
You can instead use `getRuleBasedDate(date) || defaultDate` where applicable.

Reply:
`as string` does not convert anything in typescritpt. It's just an assertion that this "is a string". You should probably do an explicit `moment(date).format(FORMAT)` instead.

Reply:
Would be great if you could also add a unit test for this function :)

Reply:
Thanks for the feedback, I'll add a unit test and I prefer @mjulstein example, `getRuleBasedDate(date) || defaultDate`, so I'll rewrite the code.

Reply:
Regarding the test, @haakemon , there is no existing test file (atleast I could not find one) and validation is quite a big file with different functions. So in my opinion the test file should isolate each function under test in a jest describe, example:

`
import 'jest';
import { getRuleBasedDate } from '../../src/utils/validation';

describe('>>> /utils/validation.ts', () => {
  describe('+++ getRuleBasedDate', () => {
    it('+++ getRuleBasedDate should return null if date is undefined', () => {
      expect(getRuleBasedDate(undefined)).toBeNull();
    });
  });
});`

Not sure what the +++ and >>> does, but I'm guessing its for generating documentation or something, so unsure of which the inner describe should use.

Reply:
Thanks for looking into that.

We have room for improvements in our tests :) I'm going to try and focus getting it in a bit better shape, and make it easier to run, but theres no quick fix for it, so it will take some time to get it in shape...